string msg
---
# Success if empty
string error
